src/lib/sql/sql_translator.cpp

```diff
@@ -934,12 +936,33 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_predicate_expression(
 
       switch (logical_expression->logical_operator) {
         case LogicalOperator::And: {
-          current_node = _translate_predicate_expression(logical_expression->right_operand(), current_node);
-          return _translate_predicate_expression(logical_expression->left_operand(), current_node);
+          current_node =
+              _translate_predicate_expression(logical_expression->right_operand(), current_node, or_translation_mode);
+          return _translate_predicate_expression(logical_expression->left_operand(), current_node, or_translation_mode);
```

Comment:
Why does `LogicalOperator::And` have an `or_translation_mode`?

Reply:
because the argument to AND could still be an OR, as in `(a OR b) AND c`